### PR TITLE
docs: add user guide for kafka remote wal

### PR DESCRIPTION
**Pull request by niebayes**
## What's Changed in this PR

Add user guide for Kafka Remote WAL.
A subsequent PR will copy the content to v0.6 after this PR is merged.

## Checklist

- [x] Please ensure that the content in `summary.yml` matches the current document structure when you changed the document structure.
- [ ] This change requires follow-up update in localized docs.

### CODE REVIEW

File: docs/v0.5/en/user-guide/operations/kafka_remote_wal.md [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+
+# How to use Kafka Remote WAL
+
+## Step 1: Start Kafka Cluster
+
+If you have already deployed a Kafka cluster, you can skip this step. However, please pay attention to the [advertised listeners](https://www.conduktor.io/kafka/kafka-advertised-host-setting/) configured during deployment, as you will use them in Step 2.
+
+
+
+We recommend using [docker compose](https://docs.docker.com/compose/) to start the Kafka cluster. Kafka supports two deployment modes: [KRaft](https://www.conduktor.io/kafka/kafka-kraft-mode/) and [Zookeeper](https://www.conduktor.io/kafka/zookeeper-with-kafka/). You can find docker compose scripts for KRaft and Zookeeper modes [here](https://github.com/confluentinc/kafka-images/blob/master/examples/confluent-server-kraft/docker-compose.yml) and [here](https://github.com/conduktor/kafka-stack-docker-compose) respectively. We advise deploying in KRaft mode, similar to the [docker-compose-standalone.yml](https://github.com/GreptimeTeam/greptimedb/blob/main/tests-integration/fixtures/kafka/docker-compose-standalone.yml) script we use, which is also pasted below for your convenience.
```
Comment (nicecui) [inline docs/v0.5/en/user-guide/operations/kafka_remote_wal.md:17]:
```suggestion
We recommend using [docker compose](https://docs.docker.com/compose/) to start the Kafka cluster. Kafka supports two deployment modes: [KRaft](https://www.conduktor.io/kafka/kafka-kraft-mode/) and [Zookeeper](https://www.conduktor.io/kafka/zookeeper-with-kafka/). You can find docker compose scripts for KRaft and Zookeeper modes [here](https://github.com/confluentinc/kafka-images/blob/master/examples/confluent-server-kraft/docker-compose.yml) and [here](https://github.com/conduktor/kafka-stack-docker-compose) respectively. We recommend deploying in KRaft mode, similar to the [docker-compose-standalone.yml](https://github.com/GreptimeTeam/greptimedb/blob/main/tests-integration/fixtures/kafka/docker-compose-standalone.yml) script we use, which is also pasted below for your convenience.
```

File: docs/v0.5/en/user-guide/operations/kafka_remote_wal.md [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+version: '3.8'
+services:
+  kafka:
+    image: bitnami/kafka:3.6.0
+    container_name: kafka
+    ports:
+      - 9092:9092
+    environment:
+      # KRaft settings
+      KAFKA_KRAFT_CLUSTER_ID: Kmp-xkTnSf-WWXhWmiorDg
```
Comment (nicecui) [inline docs/v0.5/en/user-guide/operations/kafka_remote_wal.md:29]: Where is the id from? Could users use it directly?

**Review by nicecui — COMMENTED** [review]
👍 LGTM, just need a few changes

File: docs/v0.5/en/user-guide/operations/kafka_remote_wal.md (hunk 2 above)
Comment (niebayes) [reply docs/v0.5/en/user-guide/operations/kafka_remote_wal.md:29]: Yes, users could use the script directly without worrying about the id.

**Review by nicecui — APPROVED** [review]

**Review by Wenjie0329 — APPROVED** [review]

**Pull request merged**